internal/collector/depsdev/source_test.go

```diff
@@ -0,0 +1,184 @@
+package depsdev
+
+import (
+	"context"
+	"net/url"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"go.uber.org/zap"
+
+	"github.com/ossf/criticality_score/internal/collector/signal"
+	mocks "github.com/ossf/criticality_score/internal/mock"
+)
+
+func Test_parseRepoURL(t *testing.T) {
+	tests := []struct {
+		name            string
+		u               *url.URL
+		wantProjectName string
+		wantProjectType string
+	}{
+		{
+			name: "github.com",
+			u: &url.URL{
+				Host: "github.com",
+			},
+			wantProjectName: "",
+			wantProjectType: "GITHUB",
+		},
+		{
+			name: "random",
+			u: &url.URL{
+				Host: "random",
+			},
+			wantProjectName: "",
+			wantProjectType: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotProjectName, gotProjectType := parseRepoURL(tt.u)
+			if gotProjectName != tt.wantProjectName {
+				t.Errorf("parseRepoURL() gotProjectName = %v, want %v", gotProjectName, tt.wantProjectName)
+			}
+			if gotProjectType != tt.wantProjectType {
+				t.Errorf("parseRepoURL() gotProjectType = %v, want %v", gotProjectType, tt.wantProjectType)
+			}
+		})
+	}
+}
+
+func Test_depsDevSource_Get(t *testing.T) {
+	type args struct {
+		ctx       context.Context
+		rHostName string
+		jobID     string
+	}
+	tests := []struct { //nolint:govet
```

Comment:
Just wondering if you intend to add more test cases here?

If not then the struct, the loop and the `t.Run` call are not necessary.

Reply:
I don't currently intend to make more tests for this function so I have removed the loop and `t.Run`.